platform/xbox/functions/math/pow.c

```diff
@@ -2,15 +2,48 @@
 
 double pow(double x, double y)
 {
-    return exp(y*log(x));
+    if (x < 0.0 && y != trunc(y)) {
+        // Negative base and non-integer exponent
+        return -NAN;
+    }
+
+    float r = exp(y * log(fabs(x)));
+
+    if (x < 0.0 && remainder(y, 2.0) != 0) {
+        r = copysign(r, x);
+    }
+
+    return r;
 }
 
 float powf(float x, float y)
 {
-    return expf(y*logf(x));
+    if (x < 0.0f && y != truncf(y)) {
+        // Negative base and non-integer exponent
+        return -NAN;
+    }
+
+    float r = expf(y * logf(fabsf(x)));
+
+    if (x < 0.0f && remainderf(y, 2.0f) != 0) {
```

Comment:
I'm not sure if `remainderf` is a good idea. I worry that it's potentially slow.
It appears to do an internal loop or division, so `fprem1` seems to be one of the slowest x87 instructions.

While this should rarely affect us (because most of the time `x > 0.0`), I wonder how other implementations handle it.

Reply:
Most code I have seen is derived from [Sun's implementation](https://gitlab.com/sortix/sortix/-/blob/master/libm/src/e_pow.c), which involves tons of non-trivial C code. I'm pretty sure it's more accurate, but it's crazy complex (and comes with a custom license note).

I'm open to improving this function in the future, but I'm not too worried about it being too slow rn.